apps/api/src/domain/models/user-management/group/commands/add-user-to-group/add-user-to-group.command.integration.spec.ts

```diff
@@ -0,0 +1,205 @@
+import TestRepositoryProvider from '../../../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import buildTestData from '../../../../../../test-data/buildTestData';
+import { DTO } from '../../../../../../types/DTO';
+import { IIdManager } from '../../../../../interfaces/id-manager.interface';
+import buildInMemorySnapshot from '../../../../../utilities/buildInMemorySnapshot';
+import AggregateNotFoundError from '../../../../shared/common-command-errors/AggregateNotFoundError';
+import { assertCommandError } from '../../../../__tests__/command-helpers/assert-command-error';
+import { assertCommandFailsDueToTypeError } from '../../../../__tests__/command-helpers/assert-command-payload-type-error';
+import { assertCommandSuccess } from '../../../../__tests__/command-helpers/assert-command-success';
+import { assertExternalStateError } from '../../../../__tests__/command-helpers/assert-external-state-error';
+import { DummyCommandFSAFactory } from '../../../../__tests__/command-helpers/dummy-command-fsa-factory';
+import { generateCommandFuzzTestCases } from '../../../../__tests__/command-helpers/generate-command-fuzz-test-cases';
+import { CommandAssertionDependencies } from '../../../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import { UserDoesNotExistError } from '../../errors/external-state-errors/UserDoesNotExistError';
+import { AddUserToGroup } from './add-user-to-group.command';
+import { AddUserToGroupCommandHandler } from './add-user-to-group.command-handler';
+
+const commandType = 'ADD_USER_TO_GROUP';
+
+const existingUserAlreadyInGroupId = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dc001';
+
+const existingGroupId = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dc002';
+
+const existingUserToAddToGroupId = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dc003';
+
+const userAlreadyInGroup = buildTestData().users[0].clone({ id: existingUserAlreadyInGroupId });
+
+const userToAdd = userAlreadyInGroup.clone({
+    id: existingUserToAddToGroupId,
+    username: 'al-b-added-2022',
+    authProviderUserId: 'zauth|909878',
+});
+
+const existingGroup = buildTestData().userGroups[0].clone({
+    userIds: [userAlreadyInGroup.id],
+    id: existingGroupId,
+});
+
+const initialState = buildInMemorySnapshot({
+    users: [userAlreadyInGroup, userToAdd],
+    userGroups: [existingGroup],
+});
+
+const validCommandFSA = {
+    type: commandType,
+    payload: {
+        groupId: existingGroupId,
+        userId: existingUserToAddToGroupId,
+    },
+};
+
+const buildValidCommandFSA = (): FluxStandardAction<DTO<AddUserToGroup>> => validCommandFSA;
+
+const buildInvalidFSA = (id, payloadOverrides) =>
+    new DummyCommandFSAFactory(buildValidCommandFSA).buildInvalidFSA(id, payloadOverrides);
+
+describe('AddUserToGroup', () => {
+    let app: INestApplication;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    // let arangoConnectionProvider: ArangoConnectionProvider;
```

Comment:
Remove?

Reply:
No this is important. We need to drop the dummy database in the `afterAll`. I just have to figure out why the `afterAll` happened before all the tests completed.

Reply:
fixed